app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java

```diff
@@ -168,7 +168,9 @@ public void queryPairedDevices() {
 
         this.pairedDevices = this.bluetoothAdapter.getBondedDevices();
         for (final BluetoothDevice device : this.pairedDevices) {
-            this.pairedDevicesArrayAdapter.add(device.getName() + "\n" + device.getAddress());
+            if(this.pairedDevicesArrayAdapter.getPosition(device.getName() + "\n" + device.getAddress()) == -1) {
+                this.pairedDevicesArrayAdapter.add(device.getName() + "\n" + device.getAddress());
```

Comment:
If you store the value of String.format in a final variable, you can just access that variable